theseus/geometry/tests/test_so3.py

```diff
@@ -20,11 +20,70 @@ def _create_random_so3(batch_size, rng):
     return th.SO3(quaternion=q)
 
 
-# def test_inverse():
-#     rng = torch.Generator()
-#     for batch_size in [1, 20, 100]:
-#         so3 = _create_random_so3(batch_size, rng)
-#         check_inverse(so3)
+def check_SO3_log_map(tangent_vector, group_cls):
+    error = (tangent_vector - group_cls.exp_map(tangent_vector).log_map()).norm(dim=1)
+    error = torch.minimum(error, (error - 2 * np.pi).abs())
```

Comment:
I just noticed this part. Is this necessary (taking the min with error - 2pi)? Just want to make sure this is not hiding a bug.

Reply:
I think we had similar errors when implementing one of the 2D factors, but in the end we didn't end up needing to handle this special case. Unfortunately, I wasn't able to find the PR where we had this conversation. Maybe @mhmukadam remembers? 

Two reasons why this doesn't happen in the 2D case are that are:

- your tests are more stringent, so that may explain it (btw, I would like to add similar test for 2D)
- in SE2, the approximation is triggered when |cos(theta)| ~ 1, which include -pi, 0, and pi. Maybe we can do something similar here? (where sel2 is defined, I think) If it works, this may help with the issues around `float32`?

Reply:
Actually, nevermind, I just recalled that the reason we don't trigger similar errors in SE2 is because we are using float64 in the tests, otherwise we have to handle `2pi` manually as well (I just confirmed this by changing to float32).